File: src/main/java/co/gatedaccess/web/data/repo/JoinCommunityRequestRepo.java
package co.gatedaccess.web.data.repo;

import co.gatedaccess.web.data.model.JoinCommunityRequest;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface JoinCommunityRequestRepo extends MongoRepository<JoinCommunityRequest, String> {

    JoinCommunityRequest findJoinCommunityRequestByMember_IdAndAcceptedAtIsNull(String memberId);
    Boolean existsJoinCommunityRequestByMember_IdAndAcceptedAtIsNotNull(String memberId);
    JoinCommunityRequest findJoinCommunityRequestById(String id);
}

File: src/main/kotlin/co/gatedaccess/web/data/repo/JoinCommunityRequestRepo.kt
package co.gatedaccess.web.data.repo

import co.gatedaccess.web.data.model.JoinCommunityRequest
import org.springframework.data.mongodb.repository.MongoRepository

interface JoinCommunityRequestRepo : MongoRepository<JoinCommunityRequest?, String?> {
    fun findJoinCommunityRequestByMemberIdAndAcceptedAtIsNull(memberId: String?): JoinCommunityRequest?
    fun existsJoinCommunityRequestByMemberIdAndAcceptedAtIsNotNull(memberId: String?): Boolean
    fun findJoinCommunityRequestById(id: String?): JoinCommunityRequest?
}
